### PR TITLE
Add OrderDao

### PR DESCRIPTION
**Pull request by vadymtsehelny**

### CODE REVIEW

**Review by Zeicks — APPROVED** [review]

**Review by martadiachok — APPROVED** [review]

File: src/test/java/com/easyrest/tests/administrator/AdministratorTest.java [hunk 1]
```diff
@@ -30,18 +31,24 @@ public void signIn() {
 
     @Test
     public void acceptWaitingForConfirmOrder() {
+        OrderDao orderDao = new OrderDao();
+        int insertedOrderId = orderDao.saveWaitingForConfirmOrder(email);
         Integer ordersCountBeforeAccepting = administrator.seeWaitingForConfirmOrdersCount();
         administrator.acceptOrder();
         Integer ordersCountAfterAccepting = administrator.seeWaitingForConfirmOrdersCount();
+        orderDao.deleteOrder(insertedOrderId);
```
Comment (dmatsev) [inline src/test/java/com/easyrest/tests/administrator/AdministratorTest.java]: why do you delete an order at this point and not in the teardown method?

File: src/test/java/com/easyrest/tests/administrator/AdministratorTest.java [hunk 2]
```diff
@@ -30,18 +31,24 @@ public void signIn() {
...
     }
 
     @Test
     public void assignWaiterForAcceptedOrder() {
+        OrderDao orderDao = new OrderDao();
+        int insertedOrderId = orderDao.saveAcceptedOrder(email);
         Integer ordersCountBeforeAssigning = administrator.seeAcceptedOrdersCount();
         administrator.assignWaiter();
         Integer ordersCountAfterAssigning = administrator.seeAcceptedOrdersCount();
+        orderDao.deleteOrder(insertedOrderId);
```
Comment (dmatsev) [inline src/test/java/com/easyrest/tests/administrator/AdministratorTest.java]: why do you delete an order at this point and not in the teardown method?

File: src/main/java/com/easyrest/dao/DBManager.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+public class DBManager {
+    private static final String host = ConfigProvider.host;
+    private static final String user = ConfigProvider.user;
+    private static final String password = ConfigProvider.password;
+
+    public static Connection openConnection() {
+        try {
+            return DriverManager.getConnection(host, user, password);
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
```
Comment (dmatsev) [inline src/main/java/com/easyrest/dao/DBManager.java]: why did you throw another type of exception?

File: src/main/java/com/easyrest/dao/OrderDao.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        return insertedOrderId;
+    }
+
+    public void deleteOrder(int id) {
+        try (Connection connection = DBManager.openConnection();
+             PreparedStatement deleteOrderStatement = connection.prepareStatement(DELETE_ORDER_QUERY)) {
+            deleteOrderStatement.setInt(1, id);
+            deleteOrderStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
```
Comment (dmatsev) [inline src/main/java/com/easyrest/dao/OrderDao.java]: why did you throw another type of exception?

File: src/main/java/com/easyrest/dao/OrderDao.java [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        int administratorId = 0;
+        int restaurantId = 0;
+        try (PreparedStatement administratorIdStatement = connection.prepareStatement(USER_ID_BY_EMAIL_QUERY)) {
+            administratorIdStatement.setString(1, email);
+            ResultSet administratorResultSet = administratorIdStatement.executeQuery();
+            while (administratorResultSet.next()) {
+                administratorId = administratorResultSet.getInt("id");
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
```
Comment (dmatsev) [inline src/main/java/com/easyrest/dao/OrderDao.java]: why did you throw another type of exception?

File: src/main/java/com/easyrest/dao/OrderDao.java [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+        }
+        try (PreparedStatement restaurantIdStatement = connection.prepareStatement(RESTAURANT_ID_BY_ADMINISTRATOR_ID_QUERY)) {
+            restaurantIdStatement.setInt(1, administratorId);
+            ResultSet restaurantResultSet = restaurantIdStatement.executeQuery();
+            while (restaurantResultSet.next()) {
+                restaurantId = restaurantResultSet.getInt("id");
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
```
Comment (dmatsev) [inline src/main/java/com/easyrest/dao/OrderDao.java]: why did you throw another type of exception?

File: src/main/java/com/easyrest/dao/OrderDao.java [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+            int restaurantId = getRestaurantId(connection, email);
+            try (PreparedStatement addOrderStatement = connection.prepareStatement(INSERT_ACCEPTED_ORDER_QUERY)) {
+                addOrderStatement.setInt(1, restaurantId);
+                ResultSet addOrderResultSet = addOrderStatement.executeQuery();
+                while (addOrderResultSet.next()) {
+                    insertedOrderId = addOrderResultSet.getInt(1);
+                }
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
```
Comment (dmatsev) [inline src/main/java/com/easyrest/dao/OrderDao.java]: should you stop test execution at this point?

File: src/main/java/com/easyrest/dao/OrderDao.java [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+            int restaurantId = getRestaurantId(connection, email);
+            try (PreparedStatement addOrderStatement = connection.prepareStatement(INSERT_WAITING_FOR_CONFIRM_ORDER_QUERY)) {
+                addOrderStatement.setInt(1, restaurantId);
+                ResultSet addOrderResultSet = addOrderStatement.executeQuery();
+                while (addOrderResultSet.next()) {
+                    insertedOrderId = addOrderResultSet.getInt(1);
+                }
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
```
Comment (dmatsev) [inline src/main/java/com/easyrest/dao/OrderDao.java]: should you stop test execution at this point?

**Review by dmatsev — COMMENTED** [review]
I left the comments.

File: src/test/java/com/easyrest/tests/administrator/AdministratorTest.java (hunk 1 above)
Comment (vadymtsehelny) [reply src/test/java/com/easyrest/tests/administrator/AdministratorTest.java]: Fixed. I have moved the order deleting to the teardown method.

File: src/test/java/com/easyrest/tests/administrator/AdministratorTest.java (hunk 2 above)
Comment (vadymtsehelny) [reply src/test/java/com/easyrest/tests/administrator/AdministratorTest.java]: Fixed. I have moved the order deleting to the teardown methods.

File: src/main/java/com/easyrest/dao/DBManager.java (hunk 3 above)
Comment (vadymtsehelny) [reply src/main/java/com/easyrest/dao/DBManager.java]: Fixed, removed catch block and add method throws.

File: src/main/java/com/easyrest/dao/OrderDao.java (hunk 4 above)
Comment (vadymtsehelny) [reply src/main/java/com/easyrest/dao/OrderDao.java]: threw the unchecked exception to keep the code clean

File: src/main/java/com/easyrest/dao/OrderDao.java (hunk 5 above)
Comment (vadymtsehelny) [reply src/main/java/com/easyrest/dao/OrderDao.java]: Fixed, removed catch block and add method throws.

File: src/main/java/com/easyrest/dao/OrderDao.java (hunk 6 above)
Comment (vadymtsehelny) [reply src/main/java/com/easyrest/dao/OrderDao.java]: Fixed, removed catch block and add method throws.

File: src/main/java/com/easyrest/dao/OrderDao.java (hunk 7 above)
Comment (vadymtsehelny) [reply src/main/java/com/easyrest/dao/OrderDao.java]: yes, fixed.

File: src/main/java/com/easyrest/dao/OrderDao.java (hunk 8 above)
Comment (vadymtsehelny) [reply src/main/java/com/easyrest/dao/OrderDao.java]: yes, fixed.

**Review by dmatsev — APPROVED** [review]

**Pull request merged**